OpenRA.Mods.Common/Warheads/CreateEffectWarhead.cs

```diff
@@ -107,10 +84,12 @@ public override void DoImpact(Target target, WarheadArgs args)
 			var firedBy = args.SourceActor;
 			var pos = target.CenterPosition;
 			var world = firedBy.World;
-			var targetTile = world.Map.CellContaining(pos);
-			var isValid = IsValidImpact(pos, firedBy);
+			var actorAtImpact = ImpactActors ? ActorTypeAtImpact(world, pos, firedBy) : ImpactActorType.None;
 
-			if ((!world.Map.Contains(targetTile)) || (!isValid))
+			// If there's either a) an invalid actor, or b) no actor and invalid terrain, we don't trigger the effect(s).
```

Comment:
Style nit: This comment is a bit misleading IMO (an invalid actor is allowed provided there is also a valid actor), and the `else` in the `else if` is redundant.

How about writing this as:
```csharp
// Ignore the impact if there are only invalid actors within range
if (actorAtImpact == ImpactActorType.Invalid)
	return;

// Ignore the impact if there are no valid actors and no valid terrain
// (impacts are allowed on valid actors sitting on invalid terrain!)
if (actorAtImpact == ImpactActorType.None && !IsValidAgainstTerrain(world, pos))
	return;
```
which is exactly the same code, but IMO presented a bit more clearly?